railties/lib/rails/command/behavior.rb

```diff
@@ -0,0 +1,123 @@
+require "active_support"
+
+module Rails
+  module Command
+    module Behavior #:nodoc:
+      extend ActiveSupport::Concern
+
+      class_methods do
+        # Remove the color from output.
+        def no_color!
+          Thor::Base.shell = Thor::Shell::Basic
+        end
+
+        # Track all command subclasses.
+        def subclasses
+          @subclasses ||= []
+        end
+
+        protected
+
+          # This code is based directly on the Text gem implementation.
+          # Copyright (c) 2006-2013 Paul Battley, Michael Neumann, Tim Fletcher.
+          #
+          # Returns a value representing the "cost" of transforming str1 into str2.
+          def levenshtein_distance(str1, str2)
```

Comment:
Spoke with the team and we're not gonna provide this as a public API. But maybe you can try the did_you_mean gem? It's bundled with Ruby 2.3 and also includes this code: https://github.com/yuki24/did_you_mean/blob/c22ef694f411ed4f656a029630847129c78c4cb1/lib/did_you_mean/levenshtein.rb